src/cpp/session/modules/SessionGit.cpp

```diff
@@ -485,21 +485,16 @@ class Git : public boost::noncopyable
       // build shell arguments
       ShellArgs arguments;
       
-      // on some platforms, git will return paths which contain characters
-      // not in the ASCII set with a so-called 'quoted octal encoding'.
-      // this is controlled by the 'core.quotepath' configuration option;
-      // by setting this to off we ensure that git will return us a
-      // plain UTF-8 encoded path which requires no further processing
-      arguments << "-c" << "core.quotepath=off"
-                << "status" << "--porcelain" << "--" << dir;
+      arguments << "status" << "-z" << "--porcelain" << "--" << dir;
```

Comment:
Note that we need to include both `--porcelain` and `-z` for compatibility with older versions of Git (tested with Git 1.7.1 on CentOS6)